SOURCES/etc/xapi.d/plugins/smartctl.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+import json
+import sys
+import subprocess
+import XenAPIPlugin
+
+sys.path.append('.')
+from xcpngutils import configure_logging, run_command, error_wrapped
+from xcpngutils.operationlocker import OperationLocker
+
+_LOGGER = configure_logging('smartctl')
+
+def list_disks():
+    disks = []
+    try:
+        output = subprocess.check_output(['sudo', 'smartctl', '--scan'], universal_newlines=True)
+        lines = output.split('\n')
+        for line in lines:
+            if line.startswith('/dev/') and not line.startswith('/dev/bus/'):
+                disks.append(line.split()[0])
+    except subprocess.CalledProcessError:
+        print('Error: smartctl command failed.')
+    return disks
+
+@error_wrapped
+def check_smartctl(a,b):
+    with OperationLocker():
+        results = {}
+        disks = list_disks()
+        for disk in disks:
+            cmd = ["smartctl", "-j", "-a", disk]
+            try:
+                output = subprocess.check_output(cmd)
+                json_output = json.loads(output)
+                results[disk] = json.loads(output)
+            except subprocess.CalledProcessError as e:
+                results[disk] = "Error: " + str(e)
+        return json.dumps(results)
+
+def check_health(a,b):
+    with OperationLocker():
+        results = {}
+        disks = list_disks()
+        for disk in disks:
+            cmd = ["smartctl", "-j", "-H", disk]
+            try:
+                output = subprocess.check_output(cmd)
+                json_output = json.loads(output)
+                if json_output['smart_status']['passed']:
+                    results[disk] = "PASSED"
+                else:
+                    results[disk] = "FAILED"
+            except subprocess.CalledProcessError as e:
+                results[disk] = "Error: " + str(e)
+        return json.dumps(results)
+
+
+if __name__ == "__main__":
+    XenAPIPlugin.dispatch({
+        'check_smartctl': check_smartctl,
```

Comment:
Maybe rename it to "information" to stay close to the man page. I think it will be easier to remember and the command line becomes: `... plugin=smartctl.py fn=information` that is related to what `-a` option is reported.